test/unit/app/renderer/components/navigation/navigatorTest.js

```diff
@@ -59,12 +59,32 @@ describe('Main component unit tests', function () {
     }]
   })
 
+  const customTitlebar = {
+    enabled: false,
+    captionButtonsVisible: false,
+    menubarVisible: false,
+    menubarTemplate: null,
+    menubarSelectedIndex: undefined,
+    contextMenuSelectedIndex: null,
+    lastFocusedSelector: undefined,
+    isMaximized: false
+  }
+
+  const activeTab = appState.getIn(['tabs', 0])
+
   describe('when user has history going forwards and backwards', function () {
     let wrapper
 
     before(function () {
       wrapper = shallow(
-        <Main windowState={windowState} appState={appState} />
+        <Navigator
+          windowState={windowState}
```

Comment:
passing in the entire window and app state as props will cause the Navigator component to re-render every time anything changes in either one of these stores. We need to make these props much more fine grained so they only update when they actually need to